src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -403,13 +313,24 @@ internal void PropagateOptions(CacheEntry parent)
                 }
             }
 
-            if (_absoluteExpiration.HasValue)
+            if (AbsoluteExpiration.HasValue)
             {
-                if (!parent._absoluteExpiration.HasValue || _absoluteExpiration < parent._absoluteExpiration)
+                if (!parent.AbsoluteExpiration.HasValue || AbsoluteExpiration < parent.AbsoluteExpiration)
                 {
-                    parent._absoluteExpiration = _absoluteExpiration;
+                    parent.AbsoluteExpiration = AbsoluteExpiration;
                 }
             }
         }
+
+        private void Set(State option, bool value) => _state = value ? (_state | option) : (_state & ~option);
```

Comment:
Is it valid / possible for multiple threads to be racing to set different options?  Previously as individual bools, setting one value wouldn't negatively influence another, but with this change, setting one flag could overwrite another flag being set by another thread.

Reply:
@stephentoub great catch! It's very unlikely but possible. I've pushed a fix and made it thread-safe.

Stephen, is this the reason why there is no method like `Enum.SetFlag` (while there is `Enum.HasFlag`)?

Reply:
> Stephen, is this the reason why there is no method like Enum.SetFlag (while there is Enum.HasFlag)?

I don't think so; there are plenty of APIs that perform mutations in non-atomic ways.  There's an issue about SetFlag at https://github.com/dotnet/runtime/issues/14084.